helpers/isChatToYourself.js

```diff
@@ -0,0 +1,29 @@
+const {User} = require('../databases/models');
+const UsersFunction = require('../databases/functions/users');
+
+module.exports = {
+  isChatToYourself: async (my_user_id, member) => {
+    if (my_user_id === member) return {success: false, message: 'Cannot chat to yourself'};
+    let user = await UsersFunction.findUserById(User, member);
+    if (!user) {
+      return {
+        success: false,
+        message: 'User not found'
+      };
+    }
+    const userIds = new Set([user.user_id]);
+    if (!user.is_anonymous) {
+      const anon_user = await UsersFunction.findAnonymousUserId(User, user.user_id);
+      userIds.add(anon_user.user_id);
+    } else {
+      const sign_user = await UsersFunction.findSignedUserId(User, user.user_id);
+      userIds.add(sign_user.user_id);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are multiple database calls here which can be expensive in terms of performance. Consider optimizing this by reducing the number of database calls. For example, you could modify your database functions to return all necessary user information in a single call.
```diff
-    let user = await UsersFunction.findUserById(User, member);
-    if (!user) {
-      return {
-        success: false,
-        message: 'User not found'
-      };
-    }
-    const userIds = new Set([user.user_id]);
-    if (!user.is_anonymous) {
-      const anon_user = await UsersFunction.findAnonymousUserId(User, user.user_id);
-      userIds.add(anon_user.user_id);
-    } else {
-      const sign_user = await UsersFunction.findSignedUserId(User, user.user_id);
-      userIds.add(sign_user.user_id);
-    }

+    let user = await UsersFunction.findUserWithRelatedIds(User, member);
+    if (!user || !user.related_ids) {
+      return {
+        success: false,
+        message: 'User not found'
+      };
+    }
+    const userIds = new Set(user.related_ids);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->